scripts/driverkit/build/build.py

```diff
@@ -118,6 +118,23 @@ def main():
             files = list(config_dir.glob("*.yaml"))
             print(f"[*] found {len(files)} files")
 
+            if not args.rebuild:
+                prefix = f"{s3_prefix}/{driverversion}/{arch}/"
+                s3_result = s3.list_objects_v2(Bucket=s3_bucket, Prefix=prefix, Delimiter='/')
+                exclude_list = []
+                for key in s3_result['Contents']:
+                    exclude_list.append(key['Key'])
+
+                while s3_result['IsTruncated']:
+                    continuation_key = s3_result['NextContinuationToken']
+                    s3_result = s3.list_objects_v2(Bucket=s3_bucket, Prefix=prefix, Delimiter="/", ContinuationToken=continuation_key)
+                    for key in s3_result['Contents']:
+                        exclude_list.append(key['Key'])
```

Comment:
can you extract the key iteration in a function so it's a bit more readable? this script started as a nice simple thing and then became complex :)

Reply:
Now it's a lot better 😄